drivers/flash/nrf_qspi_nor.c

```diff
@@ -262,6 +273,8 @@ static inline void qspi_clock_div_change(void)
 	 * before a QSPI transfer is performed.
 	 */
 	nrf_clock_hfclk192m_div_set(NRF_CLOCK, BASE_CLOCK_DIV);
+	/* Give the new divider a moment to take effect. */
+	k_busy_wait(7);
```

Comment:
this needs a better explanation, why "7", or "give it a moment" is too vague

Reply:
Addressed in https://github.com/zephyrproject-rtos/zephyr/pull/70172. I think this PR can be now as the nrfx update was actually needed for #70172.

Reply:
I have cherry-picked the updated commit from @anangl , please re-review